pvlib/modelchain.py

```diff
@@ -431,6 +437,22 @@ def desoto(self):
 
         self.dc = self.system.scale_voltage_current_power(self.dc).fillna(0)
 
+    def cec(self):
+        (photocurrent, saturation_current, resistance_series,
+         resistance_shunt, nNsVth) = (
+            self.system.calcparams_cec(self.effective_irradiance,
+                                       self.temps['temp_cell']))
+
+        self.diode_params = (photocurrent, saturation_current,
+                             resistance_series,
+                             resistance_shunt, nNsVth)
+
+        self.dc = self.system.singlediode(
+            photocurrent, saturation_current, resistance_series,
+            resistance_shunt, nNsVth)
+
+        self.dc = self.system.scale_voltage_current_power(self.dc).fillna(0)
```

Comment:
@mikofski's [_model_dispatcher](https://github.com/pvlib/pvlib-python/pull/555/files#r215348353) is looking better and better :/

Reply:
Note that LGTM picked up on the duplication in desoto/pvsyst once that PR was merged. Not sure why it's not picking it up in this PR. I'm still ambivalent about the best way to proceed. https://lgtm.com/projects/g/pvlib/pvlib-python/snapshot/66ee982a45ef3799644f79ed73f7b9ac5805b799/files/pvlib/modelchain.py?sort=name&dir=ASC&mode=heatmap&showExcluded=false#xf10a995000ac1e72:1

Reply:
I agree about `_model_dispatcher`.  But I'd like a design that also admits alternatives to  `pvsystem.singlediode` in addition to `calcparams_xxxx`.  For a future PR.

Reply:
Added `test_calcparams_cec`. I added a test condition that differentiates between `calcparams_desoto` and `calcparams_cec`.